blog/2024-02-01-py-client-v4-release/index.mdx

```diff
@@ -0,0 +1,436 @@
+---
+title: Major release - Python client (v4)
+slug: py-client-v4-release
+authors: [dirk, tommy, jp]
+date: 2024-02-01
+image: ./img/hero.png
+tags: ['release', 'engineering', 'concepts']
+description: "The new (v4) release of the Weaviate Python Client is - faster (gRPC), provides better IDE support and more type-safety, and many other developer experience improvements. Check out the new release and let us know what you think!"
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import FilteredTextBlock from '@site/src/components/Documentation/FilteredTextBlock';
+import PythonCode from '!!raw-loader!/_includes/code/client-libraries/python_v4.py';
+import ManageCollectionsCode from '!!raw-loader!/_includes/code/howto/manage-data.collections.py';
+
+![Weaviate Python Client v4 Release](./img/hero.png)
+
+The `v4` release of the Weaviate Python client library has landed, with a brand new API that is more powerful and easier to use 🎉🥳. This is a major update, as reflected by the bump in the major version number.
+
+It still includes the existing (`v3`) API for now, so you can continue to use it without any changes. But we do note and the `v3` API is considered deprecated, and will be removed in the future (estimated ~6 months).
+
+If you're keen to get on with it, just `pip` install it with:
+
+```bash
+pip install -U weaviate-client
+```
+
+But if you're curious about what's new, read on - there's a lot that we are excited about!
+
+## Key ideas
+
+The `v4` client does things a little differently than the `v3` client. Two of the key ideas where it differs are:
+
+- Collection-based interactions, and
+- Default property returns to queries.
+
+Take this query, for example:
+
+<FilteredTextBlock
+    text={PythonCode}
+    startMarker="# START HybridQueryExample"
+    endMarker="# END HybridQueryExample"
+    language="py"
+/>
+
+Here, you'll notice that the query:
+- is performed on a collection (`questions`), not the `client` itself, and
+- does not specify any properties to return.
+
+This is because the `v4` client is collection-based, and will return all properties (except `blob`s and references) by default. We think this makes the code more concise, and easier to use.
+
+### IDE assistance / autocomplete
+
+And writing the queries is now also much easier, with more IDE assistance and autocomplete. For example, here's a query that uses the `Filter` class:
+
+![Weaviate Python Client v4 Release](./img/pyv4_ide_example2_filter.gif)
```

Comment:
this is really cool, but cuts of sliiiightly to early

Reply:
Updated :)